src/main/java/io/confluent/connect/elasticsearch/jest/JestElasticsearchClient.java

```diff
@@ -351,6 +366,16 @@ private Index toIndexRequest(IndexableRecord record) {
     return req.build();
   }
 
+  private Update toUpdateRequest(IndexableRecord record, WriteMethod method) {
+    String payload = "{\"doc\":" + record.payload
+        + ", \"doc_as_upsert\":" + (method == WriteMethod.UPSERT) + "}";
```

Comment:
Since we should remove the `WriteMethod.UPDATE`, IMO we should also change this to be more straightforward:
```suggestion
        + ", \"doc_as_upsert\":true}";
```
And if we do that, let's also remove the `WriteMethod method` parameter from this method call, since will be unused. And maybe rename the method to `toUpsertRequest` to be more consistent with the approach.